### DISCUSSION TITLE
Reconsider blocking of memfd_create with mdwe

### DISCUSSION THREAD

**Discussion by rusty-snake** [netblue30/firejail#5126] [General]
We block memfd_create with mdwe because you can

```C
int fd = memfd_create(...);
write(fd, malicious_code, sizeof(malicious_code));
mmap(..., PROT_EXEC, ..., fd);
```

However you can also do

```C
int fd = open("/tmp", O_TMPFILE);
write(fd, malicious_code, sizeof(malicious_code));
mmap(..., PROT_EXEC, ..., fd);
```

you can't? Should we block creat/open(at(2)) too? Or miss I something that makes memfd_create special?

Moreover there are a lot programs (GDK) which use memfd_create but no W&X memory which has lead to the removal of mdwe from a lot profiles.

**Comment by topimiettinen** [comment]
I don't have a strong opinion here. memfd_create can be blocked separately, or maybe there could be a way to specify the method used by MDWE for each app.

Memory W^X policy should be complemented with file system W^X policy (noexec /tmp etc). Though there are better tricks which can evade that too.

Perhaps one day the (too) simple seccomp MDWE will be replaced with an improved in-kernel version: 
https://lore.kernel.org/linux-hardening/YmGUWq1Ea5koA8mt@arm.com/T/#u
